src/Reflection/ReflectionParser.php

```diff
@@ -26,7 +26,7 @@ final class ReflectionParser
     public function __construct()
     {
         $parserFactory = new ParserFactory();
-        $this->parser = $parserFactory->create(ParserFactory::PREFER_PHP7);
+        $this->parser = $parserFactory->createForNewestSupportedVersion();
```

Comment:
That's why I said I wasn't sure what to do here and why I initially used `createForVersion(PhpVersion::fromString('7.4'))` I'll let you decide what really needs to go here. 

BTW the new Rector probably needs to downgrade to PHP 7.4 since PHPStan dropped support for 7.2 and 7.3, but again not really sure

Reply:
I see, Rector require phpstan, so php 7.4 will be minimum then ;)

Reply:
I tested on rector side, with `->createForNewestSupportedVersion()`, it got error on curly array `$string{0}`

```
There was 1 error:

1) Rector\Tests\Php74\Rector\ArrayDimFetch\CurlyToSquareBracketArrayStringRector\CurlyToSquareBracketArrayStringRectorTest::test with data set #0 ('/Users/samsonasik/www/rector-...hp.inc')
PHPStan\Parser\ParserErrorsException: Syntax error, unexpected '{', expecting ';', Syntax error, unexpected '{', expecting ';', Syntax error, unexpected '{', expecting ';'

phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/src/Parser/RichParser.php:70
```

updating to :

```php
createForVersion(PhpVersion::fromString('7.0'))
```

make it working:

```
PHPUnit 10.5.38 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.12
Configuration: /Users/samsonasik/www/rector-src/phpunit.xml

.                                                                   1 / 1 (100%)

Time: 00:00.316, Memory: 62.50 MB

OK (1 test, 2 assertions)
```

see https://github.com/rectorphp/rector-src/pull/6431#issuecomment-2480577772

this needs fallback to php 8.0 tho, I will keep updating ;)

Reply:
Let me know if I can help in Danny way

Reply:
I got it, on rector side, using `->createForNewestSupportedVersion()` needs to be on default, then if it catch `PHPStan\Parser\ParserErrorsException`, fallback to use php 7 parser when doing parsing, see https://github.com/rectorphp/rector-src/pull/6431/commits/3bb03c6f3ec4918cd47e121a6657c49b6ed96111

If parsing process doesn't involve phpstan, you may need to use `PhpParser\Error` for catch and retry with `createForVersion(PhpVersion::fromString('7.0'))`, something like:

```php
 try {
            // parse with newest supported version
        } catch (ParserErrorsException) {
            // parse with php 7 parser
        }
```